spec/services/push_notification/http_push_spec.rb

```diff
@@ -69,9 +69,11 @@
             AppArtifacts.store.oidc_public_key,
             true,
             algorithm: 'RS256',
+            kid: JWT::JWK.new(AppArtifacts.store.oidc_private_key).kid,
```

Comment:
What does passing in the `kid` to decode do? Error if it does not match?

I don’t think we necessarily need this assertion, the one below would have a really clear failure if it errored